Sources/StreamChatUI/Utils/ImageLoading/ImageResultsMapper.swift

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+/// Helper component to map image results.
+struct ImageResultsMapper {
+    let results: [Result<UIImage, Error>]
+
+    init(results: [Result<UIImage, Error>]) {
+        self.results = results
+    }
+
+    /// Replace errors with placeholder images.
+    ///
+    /// - Parameter placeholderImages: The placeholder images.
+    /// - Returns: Returns an array of UIImages without errors.
+    func mapErrors(with placeholderImages: [UIImage]) -> [UIImage] {
+        var placeholderImages = placeholderImages
+        var finalImages: [UIImage] = []
+
+        for result in results {
+            switch result {
+            case let .success(image):
+                finalImages.append(image)
+            case .failure:
+                guard !placeholderImages.isEmpty else {
+                    continue
+                }
+
+                let placeholder = placeholderImages.removeFirst()
```

Comment:
Ideally we should be using removeLast here, but given the array in here it might not have much performance difference

Reply:
It's a 4 item, at most, I would not worry about this right now